### PR TITLE
[MAINTENANCE] Update pull request template

### PR DESCRIPTION
**Pull request by jcampbell**
Changes proposed in this pull request:
- Update pull request template
- Update associated contributor docs
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For nontrivial changes, we will conduct both a design review and a code review. Please tag a core team member for code review, or leave blank and we will find someone for you!

- [ ] Design Review (@jcampbell) (`#123`)
- [ ] Code Review

Thank you for submitting!

### CODE REVIEW

**Review by Aylr — APPROVED** [review]
I appreciate this explicitness.

**Review by eugmandel — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -6,9 +8,14 @@ Changes proposed in this pull request:
 
 After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.
 
-For nontrivial changes, we will conduct both a design review and a code review. Please tag a core team member for code review, or leave blank and we will find someone for you!
+For nontrivial changes, we will conduct both a design review and a code review. 
+
+Design Review will include ensuring that the PR is consistent with our relationship to the open source community and partner developers, with our software architecture and abstractions, and with our users' needs and expectations. We try to effectively conduct design review as early as possible, often well before a PR. Please find us via issues, discuss, or slack to engage, and then succinctly describe those interactions in the "design review" section below.
```
Comment (abegong) [inline .github/pull_request_template.md]: I bet we can boil this down so it's less intimidating. We don't need to block on that, though.

File: docs/contributing/contribution_checklist.rst [hunk 2]
```diff
@@ -62,7 +62,11 @@ When you’re done with your work...
 
     * Push to the remote fork of your repo.
     * Follow `these instructions <https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork>`__ to create a PR from your commit.
-    *  In the PR, choose a short title which sums up the changes that you have made, and in the body provide more details about what your changes do. Also mention the number of the issue where discussion has taken place, e.g. "Closes #123".
+    * Provide background for PR reviewers:
```
Comment (abegong) [inline docs/contributing/contribution_checklist.rst]: This text is awesome.

I'd suggest a small change in motivation:
"Provide background for reviewers so they can understand and approve your PR more quickly:"

Please break out the "Add the tag" stuff in its own bullet. Sub-bullets for the tags might make it more scannable, too.

**Review by abegong — APPROVED** [review]
Added 3 small comments. None of these is a blocker. I do think they'd be improvements.

**Pull request merged**